src/infrastructure/database/modules/sqlite/repositories/sqlite-badge-class.repository.ts

```
@@ -146,11 +186,15 @@ export class SqliteBadgeClassRepository implements BadgeClassRepository {
       const mergedBadgeClass = BadgeClass.create(mergedData);
 
       // Convert to database record
-      const record = this.mapper.toPersistence(mergedBadgeClass);
+      const record = { ...this.mapper.toPersistence(mergedBadgeClass) };
+      // Prevent overwriting immutable columns
+      delete (record as Partial<typeof record>).id;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)